test/org/zaproxy/zap/extension/pscanrules/PasswordAutocompleteScannerUnitTest.java

```diff
@@ -0,0 +1,169 @@
+                "</form></html>";
+        HttpMessage msg = createHttpMessage(body);
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void acOffInPass() throws HttpMalformedHeaderException {
+        String body = "<html>" +
+                "<form method='POST' action='dologin'>" +
+                "Name: <input name='usr' value='' type='text'>" +
+                "Pass: <input name='pw' type='password' autocomplete='off'>" +
+                "</form></html>";
+        HttpMessage msg = createHttpMessage(body);
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void acOfTypo() throws HttpMalformedHeaderException {
+        String body = "<html>" +
+                "<form method='POST' action='dologin' autocomplete='pff'>" +
+                "Name: <input name='usr' value='' type='text'>" +
+                "Pass: <input name='pw' type='password'>" +
+                "<input type='submit' value='Login'/>" +
+                "</form></html>";
+        HttpMessage msg = createHttpMessage(body);
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        assertThat(alertsRaised.size(), equalTo(1));
+        assertThat(alertsRaised.get(0).getParam(), equalTo("pw"));
+        assertThat(alertsRaised.get(0).getEvidence(), equalTo("<input name='pw' type='password'>"));
+	}
+
+	@Test
+	public void noAcAttrib() throws HttpMalformedHeaderException {
+        String body = "<html>" +
+                "<form method='POST' action='dologin'>" +
+                "Name: <input name='usr' value='' type='text'>" +
+                "Pass: <input name='pw' type='password'>" +
+                "</form></html>";
+        HttpMessage msg = createHttpMessage(body);
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        assertThat(alertsRaised.size(), equalTo(1));
+        assertThat(alertsRaised.get(0).getParam(), equalTo("pw"));
+        assertThat(alertsRaised.get(0).getEvidence(), equalTo("<input name='pw' type='password'>"));
+	}
+
+	@Test
+	public void acOffInWrongTag() throws HttpMalformedHeaderException {
+        String body = "<html>" +
+                "<form method='POST' action='dologin'>" +
+                "Name: <input name='usr' value='' type='text' autocomplete='off'>" +
+                "Pass: <input name='pw' type='password'>" +
+                "</form></html>";
+        HttpMessage msg = createHttpMessage(body);
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        assertThat(alertsRaised.size(), equalTo(1));
+        assertThat(alertsRaised.get(0).getParam(), equalTo("pw"));
+        assertThat(alertsRaised.get(0).getEvidence(), equalTo("<input name='pw' type='password'>"));
+	}
+
+	@Test
+	public void tworFormsOneAcOff() throws HttpMalformedHeaderException {
```

Comment:
These twoForm test cases are good. I'd suggest adding one more where autocomplete is off on one of the password fields instead of the form.